src/reg_init_distfiles.tcl.in

```diff
@@ -0,0 +1,54 @@
+#!@TCLSH@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+# Initialize distfiles table in registry.
+
+package require macports 1.0
+package require registry2 2.0
+
+mportinit
+
+if {[registry::metadata get distfiles_update_needed] == 1} {
+    set portfiles_dir [file join ${macports::registry.path} registry portfiles]
+
+    registry::write {
+        set installed_ports [registry::entry imaged]
+        foreach portref $installed_ports {
+            # Get mport reference
+            try -pass_signal {
+                # TODO: properly set ${filespath} to recognize patch files in tree
+                set mport [mportopen_installed [$portref name] [$portref version] [$portref revision] [$portref variants] {}]
```

Comment:
All patchfiles are added to the distfiles table on migration. I think it's because `mportopen_installed` does not have proper `${filespath}` set.